op-e2e/system_test.go

```diff
@@ -1093,19 +1164,29 @@ func TestFees(t *testing.T) {
 	cfg.DeployConfig.L2GenesisRegolithTimeOffset = nil
 	cfg.DeployConfig.L1GenesisBlockBaseFeePerGas = (*hexutil.Big)(big.NewInt(7))
 
-	sys, err := cfg.Start()
+	sys, err := cfg.Start(t)
 	require.Nil(t, err, "Error starting up system")
 	defer sys.Close()
 
 	l2Seq := sys.Clients["sequencer"]
 	l2Verif := sys.Clients["verifier"]
 	l1 := sys.Clients["l1"]
 
-	seqBackend := sys.Backends["sequencer"]
-	seqState, err := seqBackend.BlockChain().State()
-	require.Nil(t, err, "Error getting sequencer state")
+	config := &params.ChainConfig{
+		Optimism: &params.OptimismConfig{
+			EIP1559Elasticity:  cfg.DeployConfig.EIP1559Elasticity,
+			EIP1559Denominator: cfg.DeployConfig.EIP1559Denominator,
+		},
+		BedrockBlock: big.NewInt(0),
+	}
+
+	sga := &stateGetterAdapter{
+		ctx:    context.Background(),
+		t:      t,
+		client: l2Seq,
+	}
 
-	l1CostFn := types.NewL1CostFunc(seqBackend.BlockChain().Config(), seqState)
+	l1CostFn := types.NewL1CostFunc(config, sga)
```

Comment:
I've gone ahead and written the adapter above, to allow for us to use this `types.NewL1Cost` function from inside this test.  That being said... I'm quite confused as to _why_ we're using it.  To me, the goal of these tests is to confirm that the system is working as expected.  By pulling in this internal function, it seems like we are testing only that the output of this function is being propagated.  If for instance, a bug were introduced changing the behavior of this function, it seems this test would not catch it?

Reply:
There are (or at least should be) unit tests around the `NewL1Cost` function - the key aim of this test is to ensure that it's calculation is being added in at all the right places (and only added in once along the way). At least that's my understanding, I probably need to dig into it more closely to be sure I understand exactly what the aim is.

Reply:
You may well be right - for the purposes of this PR we're probably better off leaving what it tests the same and so stick with this shim.